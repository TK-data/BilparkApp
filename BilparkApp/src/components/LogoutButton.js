import React from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';
import { Button, Icon, Text } from 'native-base';

import { logout } from '../actions/auth';

const LogoutButton = ({ logoutUser }) => (
  <View>
    <Button
      iconRight
      danger
      onPress={logoutUser}
    >
      <Text>Logg ut</Text>
      <Icon name="log-out" />
    </Button>
  </View>
);


const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
});

const mapDispatchToProps = dispatch => ({
  logoutUser: () => dispatch(logout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(LogoutButton);
