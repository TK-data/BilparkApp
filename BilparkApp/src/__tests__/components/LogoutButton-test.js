import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import LogoutButton from '../../components/LogoutButton';

const initialState = {
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: '{"UserID":111,"Email":"awkdnawpidn@aiwfaf.no","CompanyID":7,"CarID":82,"Address":"Oiabfoaiwbf","Fname":"Obafiobafiob","Lname":"Oabfoaib","FuelDay":0,"FuelTime":"12-00","FuelNotification":false,"createdAt":"2018-05-20T11:30:10.000Z","updatedAt":"2018-05-20T11:30:21.000Z"}',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing LogoutButton', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <LogoutButton />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
