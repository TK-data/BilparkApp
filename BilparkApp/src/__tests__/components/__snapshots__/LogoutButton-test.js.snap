// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LogoutButton Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogoutButton
    isLoggedIn={undefined}
    logoutUser={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Button)
        danger={true}
        iconRight={true}
        onPress={[Function]}
        virtual={undefined}
      >
        <Styled(Text)
          virtual={undefined}
        >
          Logg ut
        </Styled(Text)>
        <Styled(Icon)
          name="log-out"
          virtual={undefined}
        />
      </Styled(Button)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Text)
            virtual={undefined}
          >
            Logg ut
          </Styled(Text)>,
          <Styled(Icon)
            name="log-out"
            virtual={undefined}
          />,
        ],
        "danger": true,
        "iconRight": true,
        "onPress": [Function],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logg ut",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Logg ut",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "log-out",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Button)
          danger={true}
          iconRight={true}
          onPress={[Function]}
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
            Logg ut
          </Styled(Text)>
          <Styled(Icon)
            name="log-out"
            virtual={undefined}
          />
        </Styled(Button)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Text)
              virtual={undefined}
            >
              Logg ut
            </Styled(Text)>,
            <Styled(Icon)
              name="log-out"
              virtual={undefined}
            />,
          ],
          "danger": true,
          "iconRight": true,
          "onPress": [Function],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logg ut",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Logg ut",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "log-out",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
